kroxylicious/src/main/java/io/kroxylicious/proxy/frame/BareSaslRequest.java

```diff
@@ -42,6 +42,12 @@ public boolean decodeResponse() {
         return decodeResponse;
     }
 
+    @Override
+    public boolean hasResponse() {
+        // unsure about this, BareSaslRequests aren't constructed anywhere but tests
```

Comment:
You could just push up the implementation onto a default method on the interface.